### PR TITLE
[DET-2785] docs: add agent config reference

### PR DESCRIPTION
**Pull request by sidneyw**

### CODE REVIEW

File: docs/reference/agent-config.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Master Port
+~~~~~~~~~~~
+The port of the Determined master located at ``master_address``. This can be configured
+via the ``master_port`` option.
+
+Agent Configuration
+====================
+
+- ``master_address``: The hostname or IP address of the Determined master.
+- ``master_port``: the port of the Determined master.
```
Comment (neilconway) [inline docs/reference/agent-config.txt:40]: Capitalize "the" for consistency.

File: docs/reference/agent-config.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+- ``agent_id``: The ID of this agent; defaults to the hostname of the current
+  machine. Agent IDs must be unique within a cluster.
+- ``label``: The label of this agent. Agents with labels may only run workloads with the corresponding label.
```
Comment (neilconway) [inline docs/reference/agent-config.txt:43]: "Agents with labels may only run workloads with the corresponding label." -> "An agent with a label will only be assigned workloads that have been assigned the same label."

File: docs/reference/agent-config.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- ``visible_gpus``: The GPUs that should be exposed as slots by the agent. A comma-separated list of GPUs,
```
Comment (neilconway) [inline docs/reference/agent-config.txt:44]: Can we give an example here? And/or suggest that users use `nvidia-smi` or similar tools to find this information about their GPUs?

File: docs/reference/agent-config.txt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+~~~~~~~~~~~~~~
+The address of the Determined master can be configured by setting the ``master_address``
+option.
+
+Master Port
+~~~~~~~~~~~
+The port of the Determined master located at ``master_address``. This can be configured
+via the ``master_port`` option.
+
+Agent Configuration
```
Comment (neilconway) [inline docs/reference/agent-config.txt:36]: This results in two top-level headings, both named "Agent Configuration". Rename this perhaps?

File: docs/reference/agent-config.txt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  master_address: 0.0.0.0
+  master_port: 8080
+
+
+Required Options
+----------------
+The following options are required for connecting an agent to the Determined
+master.
+
+Master Address
```
Comment (neilconway) [inline docs/reference/agent-config.txt:26]: Using sub-sections here seems a bit overly verbose. How about a bulleted list with two elements, one for address and one for port? This is also kinda redundant with the complete list of config options below.

File: docs/reference/agent-config.txt [hunk 6]
```diff
@@ -0,0 +1,45 @@
+.. _agent-configuration:
```
Comment (neilconway) [inline docs/reference/agent-config.txt:1]: Are there any doc pages where we want to add links to this new page?

File: docs/reference/agent-config.txt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Agent Configuration
+===================
+
+For both the master and the agent, configuration options may be set using a
```
Comment (neilconway) [inline docs/reference/agent-config.txt:6]: Is there a default path for the agent config file? And/or how do you specify the path to the config file when you start the agent? Can you specify an agent config file when using tools like `det-deploy`?

Comment (sidneyw) [reply docs/reference/agent-config.txt:6]: There is no default file path but I can definitely add an example using `det-deploy`

Comment (sidneyw) [reply docs/reference/agent-config.txt:6]: Update: `det-deploy` currently doesn't offer the ability to use a configuration file. I can give an example using `docker run` for passing a configuration file (which is how our agents are currently configured in environments where we need to add agent labels and other configuration).

Comment (neilconway) [reply docs/reference/agent-config.txt:6]: That sounds good to me. Would you mind filing a ticket for the det-deploy feature if there isn't one already?

Comment (sidneyw) [reply docs/reference/agent-config.txt]: https://determinedai.atlassian.net/browse/DET-2986

File: docs/reference/cluster-config.txt [hunk 8]
```diff
@@ -9,6 +9,16 @@ options. (Although values from different sources will be merged, we
 generally recommend sticking to a single source for each service to keep
 things simple.)
 
+Both the master and the agent can be configured using a configuration file by
+passing the `--config-file` command-line option as shown below.
```
Comment (neilconway) [inline docs/reference/cluster-config.txt:13]: Probably want two backticks, not one?

File: docs/reference/cluster-config.txt [hunk 9]
```diff
@@ -18,11 +28,21 @@ configuration file might contain
...
 to configure the host where the PostgreSQL database is running. The same
 thing could be done by starting the master with the
 ``DET_DB_HOST=the-db-host`` environment variable or ``--db-host
 the-db-host`` command-line option.
 
+The agent configuration file might contain
+
+.. code:: yaml
+
+  master_address: 0.0.0.0
```
Comment (neilconway) [inline docs/reference/cluster-config.txt:41]: How about `127.0.0.1` instead? Not sure that `0.0.0.0` will actually work...?

**Pull request merged**